docs/model_access_control.md

```diff
@@ -0,0 +1,652 @@
+          },
+          "created_time": 1684362035938,
+          "latest_version": 2,
+          "last_updated_time": 1684362571300,
+          "name": "model_group_test",
+          "description": "This is an example description",
+          "tags": {
+            "key1": "value1",
+            "key2": "value2"
+          }
+        }
+      }
+    ]
+  }
+}
+```
+
+
+
+## Deleting a model group
+
+A user can only delete model group if it does not have any model versions in it. Otherwise an exception will be thrown: `Cannot delete the model group when it has associated model versions`
+
+* On a model_access_control enabled cluster, only owner or users with matching backend roles can delete a restricted model group. 
+* Users can also delete any public model group if they have delete model_group API permission. 
+* For a private model group, only owner or the admin can delete them.
+* Admin can delete any model group
+* On a security/model_access_control disabled cluster, a user with delete model_group API permission can delete any model group
+
+
+#### Example request
+
+```
+DELETE _plugins/_ml/model_groups/<model_group_id>
+```
+
+#### Example response
+
+```
+{
+  "_index": ".plugins-ml-model-group",
+  "_id": "l8nnQogByXnLJ-QNpEk2",
+  "_version": 5,
+  "result": "deleted",
+  "_shards": {
+    "total": 2,
+    "successful": 1,
+    "failed": 0
+  },
+  "_seq_no": 70,
+  "_primary_term": 23
+}
+```
+
+
+
+## Registering a model version to a model group:
+
+Register model version will have same old path and all the old fields. Additionally, user should specify the `model_group_id` for registering a new version to a model group. If the model group id is null, an exception will be thrown
+
+On a model_access_control enabled cluster:
+* only owner or users who have atleast one of their backend roles matching with that of a model group's can register a new version to a restricted model group.
+* users can also register a version to any public model group if they have register model permission. 
+* For a private model group, only the owner or the admin can register a new version.
+* Admin can register model versions to any model group
+
+On a security/model_access_control disabled cluster, 
+* a user with register model permission can register a new version to any model group.
+
+For example:
+* For a model group created by User1 with IT backend_role, only user1, user2, and admin can register new versions. If user3 and user4 try to register, an exception will be thrown: `You don't have permissions to perform this operation on this model.`
+* For a private model group of User1, only User1 or admin will be able to create new versions. For any other user, above exception will be thrown.
+* For a public model group of User1, any user can access register a version.
+
+
+#### Example Request: Registering a model version via URL
+
+```
+POST /_plugins/_ml/models/_register
+
+{
+    "name": "all-MiniLM-L6-v2",
+    "version": "1.0.0",
```

Comment:
version is not needed here, right? You can explain more about version part

Reply:
So the auto incrementing of the version?